### PR TITLE
Added a basic name check on sign up

### PR DESCRIPTION
**Pull request by 0xf8f8ff**
When students sign up to the Quickfeed, they will be asked to provide a real name matching with the Canvas roster. 
The provided name will also be checked to ensure it does not contain any symbols besides chars and spaces.

### CODE REVIEW

File: public/src/components/forms/UserProfile.tsx [hunk 1]
```diff
@@ -47,7 +47,8 @@ export class UserProfile extends React.Component<IUserProfileProps, IUserProfile
         if (!this.props.userMan.isValidUser(curUser)) {
             message = <div style={{ color: "red" }}>
                 <p>To sign up, please complete the form below.</p>
-                <p>Please provide your real name, it is used to approve your lab submissions.</p>
+                <p>Please sign up with your real name, it is used to approve your lab submissions.</p>
```
Comment (meling) [inline public/src/components/forms/UserProfile.tsx:50]:
```html
<p>Use your <i>real name</i> as it appears on Canvas.</p>
<p>If your name does not match any names on Canvas, you will not be granted access.</p>
```

File: public/src/componentHelper.tsx [hunk 2]
```diff
@@ -338,3 +338,8 @@ export function getDaysAfterDeadline(deadline: Date, delivered: Date): number {
     const after =  Math.floor((delivered.valueOf() - deadline.valueOf()) / msInADay);
     return after > 0 ? after : 0;
 }
+
+export function isValidUserName(username: string): boolean {
+    const onlyCharsAndSpaces = /^[a-zA-Z\s]*$/;
+    return onlyCharsAndSpaces.test(username);
+}
```
Comment (meling) [inline public/src/componentHelper.tsx:345]: missing line break at end of file

**Comment by meling** [comment]
Do you plan to implement more features, like integrate with Canvas?

File: public/src/components/forms/UserProfile.tsx (hunk 1 above)
Comment (0xf8f8ff) [reply public/src/components/forms/UserProfile.tsx]: Updated.

File: public/src/componentHelper.tsx (hunk 2 above)
Comment (0xf8f8ff) [reply public/src/componentHelper.tsx]: Fixed.

**Comment by 0xf8f8ff** [comment]
> Do you plan to implement more features, like integrate with Canvas?

Do we have a required access to the canvas API?

**Comment by meling** [comment]
> > Do you plan to implement more features, like integrate with Canvas?
> 
> Do we have the required access to the canvas API?

Have not asked. Let's merge this first. We can implement Canvas later.

**Pull request merged**